### DISCUSSION TITLE
Is there a way to close preview window if I'm in normal mode inside that window?

### DISCUSSION THREAD

**Discussion by AlgusDark** [rmagatti/goto-preview#100] [Q&A] [answered]
I would like to close the current window (if it's a goto-preview window) by pressing `q`.

**Comment by rmagatti** [comment, marked as answer]
@AlgusDark you can use hooks to set a mapping to close the window by pressing only `q` when a goto-preview floating window is triggered

**Reply by AlgusDark** [reply]
Ohh, you mean by using `post_open_hook` and remap q in normal mode?

The problem with that is that the mapping will be there after I close the window 🤔 Right now I have a mapping to close the windows with <c-q> but it would be nice to have an example on how to make it so the mapping only lives in a goto_preview floating window, as I learned trough this plugin (thanks!) that you can move it to split windows and is pretty nice!

**Reply by rmagatti** [reply]
So, it's not possible currently to set `window-local` mappings. What you can do is in your hook, check if any goto-preview windows are open, and if not unregister the mapping you set for closing.
https://github.com/neovim/neovim/issues/16263
https://github.com/vim/vim/issues/9339
